File: baekjoon/1459_걷기/걷기.js
// 실버3 : https://www.acmicpc.net/problem/1459

const fs = require("fs");
const file = process.platform === "linux" ? "dev/stdin" : "./text.txt";
const input = fs.readFileSync(file).toString().trim().split(" ").map(Number);
let [x, y, w, s] = input;
let result = 0;

if (w * 2 <= s) {
  result = (x + y) * w;
} else {
  if (w < s) {
    result = Math.min(x, y) * s + Math.abs(x - y) * w;
  } else {
    if (Math.abs(x - y) % 2 === 0) {
      result = Math.min(x, y) * s + Math.abs(x - y) * s;
    } else {
      result = Math.min(x, y) * s + (Math.abs(x - y) - 1) * s + w;
    }
  }
}
console.log(result);

File: baekjoon/silver/3_1459_걷기/걷기.py
# 실버3 : https://www.acmicpc.net/problem/1459

x, y, w, s = map(int, input().split())
result = 0

# 무조건 가로, 세로로 이동
if w * 2 <= s:
  result = w*(x+y)

# 대각선을 최대한 많이
else:
  if w < s:
    result = min(x,y)*s + abs(x-y) * w
  else:
    if abs(x-y) % 2 == 0:
      result = min(x,y)*s + abs(x-y) * s
    else:
      result = min(x,y)*s + (abs(x-y) -1) * s + w

    
print(result)